plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java

```diff
@@ -118,6 +122,20 @@ protected IStatus doRun(IProgressMonitor monitor) {
 		return Status.OK_STATUS;
 	}
 
+	private boolean isUpdate(Connection connection, String project, String name) {
+		try {
+			IDeploymentConfig dc = connection.getResource(ResourceKind.DEPLOYMENT_CONFIG, project, name);
```

Comment:
Actually the DC can not be null; that would imply a client error.  It should return something or throw a not found exception.  I will update the code to catch an OpenShiftException, but you are correct.  I'm not certain why this NotFound was not wrappered by the public one.